CMakeLists.txt

```diff
@@ -66,76 +74,98 @@ install(DIRECTORY include/${PROJECT_NAME} DESTINATION include)
 install(EXPORT ${PROJECT_NAME}-targets
     FILE "${PROJECT_NAME}-targets.cmake"
     NAMESPACE steering_functions::
-    DESTINATION "lib/cmake/${PROJECT_NAME}"
+    DESTINATION "share/${PROJECT_NAME}/cmake"
 )
 install(
     FILES
         "${PROJECT_NAME}-config.cmake"
         "${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}-config-version.cmake"
-    DESTINATION "lib/cmake/${PROJECT_NAME}"
+    DESTINATION "share/${PROJECT_NAME}/cmake"
 )
 
 if(BUILD_WITH_ROS)
-    add_executable(${PROJECT_NAME}_node src/${PROJECT_NAME}_node.cpp)
-
-    target_link_libraries(${PROJECT_NAME}_node
-        PRIVATE
-            steering_functions
-            Eigen3::Eigen
-            ${catkin_LIBRARIES}
-    )
-
-    target_include_directories(${PROJECT_NAME}_node
-        PRIVATE
-            ${catkin_INCLUDE_DIRS}
-    )
-
-    install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_node
-        ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-        RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION}
-    )
-
-    install(DIRECTORY include/${PROJECT_NAME}/
-        DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
-    )
-
-    install(DIRECTORY
-        config
-        launch
-        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
-    )
-
-    if(CATKIN_ENABLE_TESTING)
-        ## Add gtest based cpp test target and link libraries
-        catkin_add_gtest(steering_functions_test test/steering_functions_test.cpp)
-        if(TARGET steering_functions_test)
-            target_link_libraries(steering_functions_test ${catkin_LIBRARIES} ${PROJECT_NAME})
-            target_include_directories(steering_functions_test PRIVATE ${catkin_INCLUDE_DIRS})
+    if(${ROS_VERSION} EQUAL 1)
```

Comment:
This requires `<build_depend>ros_environment</build_depend>`

Reply:
Fixed